cms/djangoapps/contentstore/views/component.py

```diff
@@ -565,6 +559,13 @@ def component_handler(request, usage_key_string, handler, suffix=''):
 
     # unintentional update to handle any side effects of handle call
     # could potentially be updating actual course data or simply caching its values
-    modulestore().update_item(descriptor, request.user.id, asides=asides)
+    # Addendum:
+    # TNL 101-62 studio write permission is also checked for editing content.
+
+    if has_course_author_access(request.user, usage_key.course_key):
+        modulestore().update_item(descriptor, request.user.id, asides=asides)
+    else:
+        #try to fail loudly
+        raise PermissionDenied("No studio write Permissions, no changes to course content made")
```

Comment:
If we're going to throw an exception when the user doesn't have permission, I think it's better to do it where you had it before, at the start of the function before the handler (`handler_descriptor.handle(...)`) actually runs. Because in this case, you'll get an error but some side effects may have actually happened within the XBlock code.

On the other hand, if we're going to allow handlers but _silently_ prevent writes for users without permission, I think this is the right place to do it, by gating the call to `update_item`.

This is a case where there is no official spec or guideliness for how this part of the XBlock API should handle authentication so we just have to pick what seems most sensible and is compatible with most existing xblocks.